src/bgw/job.c

```diff
@@ -84,22 +189,72 @@ bgw_job_from_tupleinfo(TupleInfo *ti, size_t alloc_size)
 	Assert(alloc_size >= sizeof(BgwJob));
 	job = MemoryContextAllocZero(ti->mctx, alloc_size);
 	tuple = ts_scanner_fetch_heap_tuple(ti, false, &should_free);
-	memcpy(job, GETSTRUCT(tuple), sizeof(FormData_bgw_job));
-
-	if (should_free)
-		heap_freetuple(tuple);
 
 	/*
-	 * GETSTRUCT does not work with variable length types and NULLs so we have
-	 * to do special handling for hypertable_id and the jsonb column
+	 * Using heap_deform_tuple instead of GETSTRUCT since the tuple can
+	 * contain NULL values. Some of these cannot really be null, but we check
```

Comment:
The stuff that cannot happen should be checked with Assert. Checking it with `if` gives an impression that this is a valid case, contrary to this comment.

Reply:
this was written by Mats so I'm hesitant to change it.

Reply:
Just do it quick, while he's on vacation :smile: